api/datastore/backup.go

```diff
@@ -55,24 +53,38 @@ func (store *Store) copyDBFile(from string, to string) error {
 
 // BackupOptions provide a helper to inject backup options
 type BackupOptions struct {
-	Version        int
+	Version        int // I can't find this used for anything other than a filename
 	BackupDir      string
 	BackupFileName string
 	BackupPath     string
 }
 
+// getBackupRestoreOptions returns options to store db at common backup dir location; used by:
+// - db backup prior to version upgrade
+// - db rollback
+func getBackupRestoreOptions(backupDir string) *BackupOptions {
+	return &BackupOptions{
+		BackupDir:      backupDir, //connection.commonBackupDir(),
+		BackupFileName: beforePortainerVersionUpgradeBackup,
+	}
+}
+
 func (store *Store) setupOptions(options *BackupOptions) *BackupOptions {
 	if options == nil {
 		options = &BackupOptions{}
 	}
 	if options.Version == 0 {
-		options.Version, _ = store.version()
+		version, err := store.version()
+		if err != nil {
+			version = 0
+		}
+		options.Version = version
```

Comment:
potentially changes control flow? since if no version in DB this would've previously errored/panic'd?